l10n_br_fiscal/models/res_partner.py

```diff
@@ -25,6 +25,13 @@ def _default_fiscal_profile_id(self, is_company=False):
             [('default', '=', True), ('is_company', '=', is_company)],
             limit=1)
 
+    @api.multi
+    def _compute_fiscal_document_count(self):
+        for partner in self:
+            partner.fiscal_documents_count = partner.fiscal_document_ids.search_count(
+                [('partner_id', '=', partner.id)]
```

Comment:
I don't think it's necessary to apply this domain, because the recordset is already a one2many field and contains only the documents which the partner_id field is the partner record in question. You could do the same with `len(partner.fiscal_document_ids)` or `partner.fiscal_document_ids.search_count([])`